### PR TITLE
[WIP] Add dimensions/tax code to PDP

### LINKED ISSUE #1699: PDP UX for Shipping, Taxes
For implementation of #1618,  we need UX for designed for the following information on product details page:

Dimensions [Width (x) Length, Depth]
Customs Category (select from list)
Tax Code Category  (select from list)

@rymorgan this is likely some additional fields, in the same area vein as the existing inventory, pricing.

### PR DESCRIPTION
**Pull request by zenweasel**
Resolves #1699

### CODE REVIEW

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js [hunk 1]
```diff
@@ -4,10 +4,18 @@ import { Template } from "meteor/templating";
 import { Reaction, i18next } from "/client/api";
 import { ReactionProduct } from "/lib/api";
 import { applyProductRevision } from "/lib/api/products";
-import { Products } from "/lib/collections";
+import { Products, Packages } from "/lib/collections";
+import { ReactiveDict } from "meteor/reactive-dict";
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js]: Import order

File: lib/collections/schemas/products.js [hunk 2]
```diff
@@ -276,6 +318,19 @@ export const ProductVariant = new SimpleSchema({
   workflow: {
     type: Workflow,
     optional: true
+  },
+
+  originCountry: {
```
Comment (zenweasel) [inline lib/collections/schemas/products.js]: originCountry should not be required.

basic_reaction_product

File: lib/collections/schemas/products.js [hunk 3]
```diff
@@ -138,6 +138,48 @@ export const ProductVariant = new SimpleSchema({
       }
     }
   },
+  length: {
```
Comment (zenweasel) [inline lib/collections/schemas/products.js]: Dimensions should not be required.

basic_reaction_product

Comment (kieha) [inline lib/collections/schemas/products.js]: Fixed

File: lib/collections/schemas/products.js (hunk 2 above)
Comment (kieha) [inline lib/collections/schemas/products.js]: Fixed

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js (hunk 1 above)
Comment (kieha) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js]: Fixed

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js [hunk 4]
```diff
@@ -4,10 +4,18 @@ import { Template } from "meteor/templating";
 import { Reaction, i18next } from "/client/api";
 import { ReactionProduct } from "/lib/api";
 import { applyProductRevision } from "/lib/api/products";
-import { Products } from "/lib/collections";
+import { Packages, Products } from "/lib/collections";
+import { ReactiveDict } from "meteor/reactive-dict";
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js]: Import order

File: .meteor/packages [hunk 5]
```diff
@@ -96,3 +96,5 @@ johanbrook:publication-collector
 
 
 # Custom Packages
+aldeed:autoform-select2
```
Comment (zenweasel) [inline .meteor/packages]: If we aren't using this autoform package anymore, let's remove it

File: imports/plugins/included/product-variant/server/methods/populateTaxCodes.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      id: code.id,
+      shopId: shopId,
+      taxCode: code.taxCode,
+      taxCodeProvider: providerName,
+      ssuta: code.isSSTCertified,
+      label: code.description,
+      parent: code.parentTaxCode
+    });
+  } catch (e) {
+    throw e;
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/server/methods/populateTaxCodes.js]: Let's try to throw a `Meteor.Error` with a better description of where it happened and what happened.

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js (hunk 4 above)
Comment (kieha) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js]: Fixed

File: .meteor/packages (hunk 5 above)
Comment (kieha) [inline .meteor/packages]: Fixed

**Pull request closed without merging**